pkg/graph/resolvers/system_intake_status_requester.go

```diff
@@ -0,0 +1,107 @@
+	if intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState == models.SIDSNoDecision {
+		return models.SISRClosed, nil
+	}
+
+	switch intake.Step {
+	case models.SystemIntakeStepINITIALFORM:
+		return calcSystemIntakeInitialFormStatusRequester(intake.RequestFormState)
+	case models.SystemIntakeStepDRAFTBIZCASE:
+		return calcSystemIntakeDraftBusinessCaseStatusRequester(intake.DraftBusinessCaseState)
+	case models.SystemIntakeStepGRTMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRTMeetingStatusRequester(intake.GRTDate, currentTime), nil
+	case models.SystemIntakeStepFINALBIZCASE:
+		return calcSystemIntakeFinalBusinessCaseStatusRequester(intake.FinalBusinessCaseState)
+	case models.SystemIntakeStepGRBMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRBMeetingStatusRequester(intake.GRBDate, currentTime), nil
+	case models.SystemIntakeStepDECISION:
+		return calcSystemIntakeDecisionStatusRequester(intake.DecisionState)
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid step")
+	}
+}
+
+func calcSystemIntakeInitialFormStatusRequester(intakeFormState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch intakeFormState {
+	case models.SIRFSNotStarted:
+		return models.SISRInitialRequestFormNew, nil
+	case models.SIRFSInProgress:
+		return models.SISRInitialRequestFormInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRInitialRequestFormEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRInitialRequestFormSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid intakeFormState")
+	}
+}
+
+func calcSystemIntakeDraftBusinessCaseStatusRequester(draftBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch draftBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRDraftBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRDraftBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRDraftBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid draftBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRTMeetingStatusRequester(grtDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grtDate == nil || grtDate.After(currentTime) {
+		return models.SISRGrtMeetingReady
+	}
+
+	return models.SISRGrtMeetingAwaitingDecision
+}
+
+func calcSystemIntakeFinalBusinessCaseStatusRequester(finalBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch finalBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
```

Comment:
Shouldn't the initial state return a `READY_TO_START`?

Reply:
The overall intake status goes right to "Final Business Case in progress" ([Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=477-113294&mode=design&t=foWugeQRbZN35Xtx-0)). The task list status will have `READY_TO_START` as a possible value, that'll be in EASI-2961.

Reply:
That's weird, but OK, if that's what the designs call for. Thanks for clarifying.